ui/src/state/chat/chat.ts

```diff
@@ -1008,6 +1009,42 @@ export function useWritByFlagAndWritId(
   return cached;
 }
 
+export function useWritByFlagAndGraphIndex(
+  chFlag: string,
+  index: string,
+  isScrolling: boolean
+) {
+  const res = useChatState(
+    useCallback((s) => s.loadedGraphRefs[chFlag + index], [chFlag, index])
+  );
+
+  useEffect(() => {
+    if (!res && !isScrolling) {
+      (async () => {
+        let w: ChatWrit | 'error' = 'error';
+        try {
+          useChatState.getState().batchSet((draft) => {
+            draft.loadedGraphRefs[chFlag + index] = 'loading';
+          });
+          const { writ } = await subscribeOnce(
+            'chat',
+            `/hook/${chFlag}${index}`
+          );
+          w = writ;
+        } catch (e) {
+          console.warn(e);
+        }
+
+        useChatState.getState().batchSet((draft) => {
+          draft.loadedGraphRefs[chFlag + index] = w;
+        });
+      })();
+    }
+  }, [isScrolling, chFlag, index, res]);
+
+  return res || 'loading';
+}
```

Comment:
seems like we always use flag and index together, should probably combine into one